src/wallet/mod.rs

```diff
@@ -4097,4 +4170,167 @@ pub(crate) mod test {
             "when there's no internal descriptor it should just use external"
         );
     }
+
+    #[test]
+    fn test_taproot_psbt_populate_tap_key_origins() {
+        let (wallet, _, _) = get_funded_wallet(get_test_tr_single_sig_xprv());
+        let addr = wallet.get_address(AddressIndex::New).unwrap();
+
+        let mut builder = wallet.build_tx();
+        builder.add_recipient(addr.script_pubkey(), 25_000);
+        let (psbt, _) = builder.finish().unwrap();
+
+        assert_eq!(
+            psbt.inputs[0].tap_key_origins.len(),
+            1,
+            "Empty input tap_key_origins"
+        );
+        assert_eq!(
+            psbt.outputs[0].tap_key_origins.len(),
+            1,
+            "Empty output tap_key_origins"
+        );
```

Comment:
Instead of only checking length feels like its better to assert the data structure itself?? The descriptor is hardcoded so they should not change..

Reply:
Yes, that's a good point